src/cargo/util/lints.rs

```diff
@@ -133,18 +133,16 @@ fn verify_feature_enabled(
             dash_feature_name
         );
 
-        let message = if let Some(span) =
-            get_span(manifest.document(), &["lints", "cargo", lint_name], false)
-        {
-            Level::Error
-                .title(&title)
-                .snippet(
-                    Snippet::source(manifest.contents())
-                        .origin(&manifest_path)
-                        .annotation(Level::Error.span(span).label(&label))
-                        .fold(true),
-                )
-                .footer(Level::Help.title(&help))
+        let mut groups = Vec::new();
+        let mut group = Group::with_title(Level::ERROR.primary_title(title));
+
+        if let Some(span) = get_span(manifest.document(), &["lints", "cargo", lint_name], false) {
+            group = group.element(
+                Snippet::source(manifest.contents())
+                    .path(manifest_path)
+                    .annotation(AnnotationKind::Primary.span(span).label(label)),
+            );
+            group = group.element(Level::HELP.message(help));
```

Comment:
Aside: If we have to have a lot of this `group = group.call()` stuff, I wonder if that is a sign we should consider a `&mut self` builder for the next breaking change